src/ManageCourses.Api/Services/EnrichmentService.cs

```diff
@@ -21,22 +21,94 @@ public EnrichmentService(IManageCoursesDbContext context)
+        ///check for the existence of a previous published record and get the last pulished date
+        /// </summary>
+        /// <param name="model">holds the enriched data</param>
+        /// <param name="instCode">the institution code for the enrichment data</param>
+        /// <param name="email">the email of the user</param>
+        public void SaveInstitutionEnrichment(UcasInstitutionEnrichmentPostModel model, string instCode, string email)
+        {
+            var userInst = ValidateUserOrg(email, instCode);
+
+            var enrichmentDraftRecord = _context.InstitutionEnrichments.Where(ie => instCode.ToLower() == ie.InstCode.ToLower() && ie.Status == EnumStatus.Draft).OrderByDescending(x => x.Id).FirstOrDefault();
+            
+            var content = JsonConvert.SerializeObject(model.EnrichmentModel, _jsonSerializerSettings);
+
+            if (enrichmentDraftRecord != null)
             {
-                var enrichmentModel = enrichment.JsonData != null ? JsonConvert.DeserializeObject<InstitutionEnrichmentModel>(enrichment.JsonData, _jsonSerializerSettings) : null;
+                //update
+                enrichmentDraftRecord.UpdatedTimestampUtc = DateTime.UtcNow;
+                enrichmentDraftRecord.UpdatedByUser = userInst.McUser;
+                enrichmentDraftRecord.JsonData = content;                
+            }
+            else
+            {
+                //insert
+                var enrichmentPublishRecord = _context.InstitutionEnrichments.Where(ie => instCode.ToLower() == ie.InstCode.ToLower() && ie.Status == EnumStatus.Published).OrderByDescending(x => x.Id).FirstOrDefault();
+                DateTime? lastPublishedDate = null;
+                if (enrichmentPublishRecord != null)
+                {
+                    lastPublishedDate = enrichmentPublishRecord.LastPublishedTimestampUtc;
+                }
+                var enrichment = new InstitutionEnrichment
+                {
+                    InstCode = userInst.UcasInstitution.InstCode,
+                    CreatedTimestampUtc = DateTime.UtcNow,
+                    UpdatedTimestampUtc = DateTime.UtcNow,
+                    LastPublishedTimestampUtc = lastPublishedDate,
+                    CreatedByUser = userInst.McUser,
+                    UpdatedByUser = userInst.McUser,
+                    Status = EnumStatus.Draft,
+                    JsonData = content,
+                };
+                _context.InstitutionEnrichments.Add(enrichment);
+            }
+            _context.Save();
+        }
+        /// <summary>
+        /// Changes the status of the latest draft record to published
+        /// </summary>
+        /// <param name="instCode">institution code of the enrichemtn to be published</param>
+        /// <param name="email">email of the user</param>
+        /// <returns>true if successful</returns>
+        public bool PublishInstitutionEnrichment(string instCode, string email)
```

Comment:
Use exceptions not return values to indicate error conditions.

Reject

Reply:
The bool will never indicate an error condition. It simply indicates whether or not the publish succeeded. For example if a draft enrichment record does not exist then this returns false when is quite reasonable and not an error. There is no try/catch here so all exceptions will just throw at that point.